projects/ngx-chess-board/src/lib/ngx-chess-board.component.ts

```diff
@@ -208,19 +228,23 @@ export class NgxChessBoardComponent
         return this.engineFacade.board.fen;
     }
 
-    dragEnded(event: CdkDragEnd): void {
+    dragEnded(event: CdkDragEnd, piece: Piece): void {
         this.isDragging = false;
+
+        const lastMove = this.engineFacade.moveHistoryProvider.getLastMove();
+        const length = this.engineFacade.getMoveHistory().length;
+
         this.engineFacade.dragEndStrategy.process(
             event,
-            this.engineFacade.moveDone,
+            piece === lastMove?.piece && length !== this.dragStartHistoryLength,
```

Comment:
This was causing multiple bugs
When to detect on dragEnd if a move was done or not

1) If I dragdown a piece, and a move was done by opponent or bot, it piece I dragged down freezes

2) Comparing the last move piece with the dragged down piece:
What if you predict the next move, and you drag it down, and it really gets moved, it does get frozen
-> To resolve this, I had to implement disable drag of opponent pieces (Not elegeant but fine)